Remove InlayHint filtering by the client

Plixo2 opened the pull request:
I removed here the client-side InlayHint filtering, to add it to the server instead.

I have removed the `InlayHintsProvider`, re-enabled the `InlayHintsFeature` from the VSCode extension and filter out captures/returnTypes on the Server and everything worked fine, see [here](https://github.com/effekt-lang/effekt/compare/master...Plixo2:effekt:master)

@timsueberkrueb previously stated: 

> When just looking at the amount of code necessary, it seems to be easier to implement it on the server side: we already have the extension settings available on the server (via `initializationOptions` and `didChangeConfiguration`. This would also allow us to get rid of the `data` field and avoid hacks such as the one above.
> 
> However, from an architectural perspective, it seems to be a violation of concerns to add view-related logic to the language server (even using the extension settings on the server is an architectural smell IMHO), so from that perspective, one might argue for the hack on the client side.

Describing it as a violation of separation of concerns might be valid, but:
1. The configuration is in the LSP standard and should be used exactly for this.
2. The LSP would work properly for other clients
3. The configuration setting is already used on the server (`showTree`, `showHoles` and `showExplanations`), so just handling inlay hints on the client is the far bigger smell

Here is also what @jiribenes had to say ([here](https://github.com/se-tuebingen-epe/course/discussions/13#discussioncomment-14866950)):


<img width="1834" height="478" alt="image" src="https://github.com/user-attachments/assets/0c5a1f08-dc1f-4aa4-94b6-c4f24dab8dd3" />

---

I did confirm that `didChangeConfiguration()` does work properly, as its [automatically synchronized](https://github.com/effekt-lang/effekt-vscode/blob/7a3e69bdb66de0d60833a9722b5f7eefc63b5592/src/extension.ts#L244)

I have the corresponding PR for the LSP implementation ready to go, after this is approved. ([Diff](https://github.com/effekt-lang/effekt/compare/master...Plixo2:effekt:master))

Also i followed all steps in [CONTRIBUTING.md](https://github.com/effekt-lang/effekt-vscode/blob/master/CONTRIBUTING.md)

On a separate note, could [`compileOnChange: Boolean`](https://github.com/effekt-lang/effekt/blob/4d7605baaf0c9a5282431208fb4285e9cf04bc3b/effekt/jvm/src/main/scala/effekt/Server.scala#L34) be set to true again? The comment  states `Currently disabled because references are erased`, but it would be way better to have immediate errors and new inlay hints instead of preserving old references (At least for beginners, so a setting might be appropriate)

Plixo2 commented on package.json:
```diff
@@ -7,7 +7,7 @@
   },
   "icon": "icon.png",
-  "version": "0.6.2",
+  "version": "0.6.3",
```
Publishing appears to be done via Github workflows, without any version bump, or am i missing something here?

timsueberkrueb reviewed (changes requested):
Thanks for the PR! Before we merge this, please also create the corresponding PR on the `effekt` side. 

Note that we will have to release the change on the `effekt` side first. This will lead to the inlay hints being filtered twice, but it will preserve the current observable behavior. If we were to release the extension first, filtering inlay hints would be broken until the next `effekt` release is made, which we should avoid.

Plixo2 commented:
@timsueberkrueb I implemented everything

1. Removed the Release: 
      > Instead, following our [release process](https://github.com/effekt-lang/effekt-vscode/blob/master/CONTRIBUTING.md#release-process), we typically use npm version to generate a commit that bumps the version
      
      Yeah, i read past the `Release Process` heading and thought i had to include this. I actually used `npm version` btw. 

2. Replaced `EffektLanguageClient` by the default class. I have kept `minify: false`, as this might cause bugs in the future and is out of scope of this PR.

timsueberkrueb commented:
Thanks! I'm happy with these changes. I will approve and merge this next week once the next `effekt` release has been made (to prevent accidental breakage).
